language/e2e-tests/src/executor.rs

```diff
@@ -166,6 +168,31 @@ impl FakeExecutor {
         )
     }
 
+    /// Executes the transaction as a singleton block and applies the resulting write set to the
+    /// data store. Returns None if execution fails
+    pub fn execute_and_apply(
+        &mut self,
+        transaction: SignedTransaction,
+    ) -> Option<TransactionOutput> {
```

Comment:
we can probably panic inside this function if we don't need to test failure case or we better return Result instead of Option?